src/vs/workbench/api/electron-browser/mainThreadTerminalService.ts

```diff
@@ -155,6 +156,24 @@ export class MainThreadTerminalService implements MainThreadTerminalServiceShape
 		terminalInstance.addDisposable(this._terminalOnDidWriteDataListeners[terminalId]);
 	}
 
+	public $registerOnTitleChangedListener(terminalId: number): void {
```

Comment:
As discussed, I think we can just remove this and always send titles over from the main side.